virttest/nvme.py

```diff
@@ -0,0 +1,69 @@
+"""
+This module provides interfaces about NVMe storage backend.
+
+Available functions:
+- get_image_filename: Get the image filename from NVMe.
+- file_exists: Check whether the NVMe image file exists.
+- get_address_namespace: Get the address and namespace from NVMe image filename.
+
+"""
+import re
+
+from avocado.utils import process
+
+from virttest import utils_misc
+from virttest.compat_52lts import decode_to_text
+
+
+def get_image_filename(address, namespace):
+    """
+    Get the image filename from NVMe.
+
+    :param address: The PCI address NVMe,
+                    format: $domain:$bus:$slot.$function/$namespace.
+    :type address: str
+    :param namespace: The namespace number starting according to the NVMe spec.
+    :type namespace: str
+    :return: The image filename from NVMe,
+             the format of nmve://$domain:$bus:$slot.$function/$namespace
+             e.g: nvme://0000:44:00.0/1
+    :rtype: str
+    """
+    cmd = 'lspci -s %s' % address
+    o = decode_to_text(process.system_output(cmd, 60, False))
+    if not o:
+        raise ValueError('No found device[%s] on host.' % address)
+    addr = re.search(r'\w+:(\w+:\w+\.\w+)', address).group(1)
+    regex = r'.*%s Non-Volatile memory controller:.* NVMe .*' % addr
+    if not re.search(regex, o, re.A):
+        raise ValueError('Device[%s] is not Non-Volatile memory(NVMe) controller.'
+                         % address)
+    return 'nvme://%s/%s' % (address, namespace)
+
+
+def file_exists(params, filename):
+    """
+    Check whether the NVMe image file exists.
+
+    :param params: A dict containing image parameters.
+    :type params: dict
+    :param filename: The NVMe image filename.
+    :type filename: str
+    :return: True if the NVMe image file exists, else False
+    :rtype: bool
+    """
+    cmd = "%s %s" % (utils_misc.get_qemu_img_binary(params), filename)
```

Comment:
should this be 'lspci -s address.namespace' instead of qemu-img

Reply:
You could not use `lspci -s address.namespace`
because:
`# man lspci`
```
...
   Options for selection of devices
       -s [[[[<domain>]:]<bus>]:][<device>][.[<func>]]
...
```
so can not check the format `[[[[<domain>]:]<bus>]:][<device>][.[<func>].[namespace]]`